internal/checker/ruleset_info.go

```diff
@@ -7,43 +7,92 @@ import (
 	"path/filepath"
 )
 
-func ReadSpitoYaml(rulesetLocation *RulesetLocation) ([]byte, error) {
-	spitoRulesDataBytes, err := os.ReadFile(rulesetLocation.GetRulesetPath() + "/spito.yml")
-	if os.IsNotExist(err) {
-		var err2 error
-		spitoRulesDataBytes, err2 = os.ReadFile(rulesetLocation.GetRulesetPath() + "/spito.yaml")
-		if err2 != nil {
-			return nil, err2
+func GetRulesetConf(rulesetLocation *RulesetLocation) (RulesetConf, error) {
+	spitoRulesetConf := RulesetConf{
+		Rules: make(map[string]RuleConf),
+	}
+
+	rulesetConfYaml, err := getRulesetConfYaml(rulesetLocation)
+	if err != nil {
+		return RulesetConf{}, err
+	}
+
+	for key := range rulesetConfYaml.Rules {
+		ruleConf, err := rulesetConfYaml.GetRuleConfBasedOnYaml(rulesetLocation, key)
+		if err != nil {
+			return RulesetConf{}, err
 		}
+
+		spitoRulesetConf.Rules[key] = ruleConf
 	}
-	return spitoRulesDataBytes, err
+
+	return spitoRulesetConf, nil
 }
 
-func getRulesetConf(rulesetLocation *RulesetLocation) (RulesetConf, error) {
-	spitoRulesetYamlRaw, err := ReadSpitoYaml(rulesetLocation)
+func GetRuleConf(rulesetLocation *RulesetLocation, ruleName string) (RuleConf, error) {
+	rulesetConfYaml, err := getRulesetConfYaml(rulesetLocation)
 	if err != nil {
-		return RulesetConf{}, err
+		return RuleConf{}, err
 	}
 
-	var spitoRulesetYaml SpitoRulesetYaml
-	if err := yaml.Unmarshal(spitoRulesetYamlRaw, &spitoRulesetYaml); err != nil {
-		return RulesetConf{}, err
+	return rulesetConfYaml.GetRuleConfBasedOnYaml(rulesetLocation, ruleName)
+}
+
+func GetRuleConfFromScript(scriptPath string) (RuleConf, error) {
+	ruleConf := RuleConf{
+		Path:        scriptPath,
+		Unsafe:      false,
+		Environment: false,
 	}
 
-	spitoRulesetConf := RulesetConf{
-		Rules: make(map[string]RuleConf),
+	scriptRaw, err := os.ReadFile(scriptPath)
+	if err != nil {
+		return RuleConf{}, err
 	}
 
-	for key := range spitoRulesetYaml.Rules {
-		ruleConf, err := spitoRulesetYaml.getRuleConf(key)
-		if err != nil {
-			return RulesetConf{}, err
+	// Get data from decorators
+	processScript(string(scriptRaw), &ruleConf)
+
+	return ruleConf, err
+}
+
+func (s *RulesetConfYaml) GetRuleConfBasedOnYaml(rulesetLocation *RulesetLocation, ruleName string) (RuleConf, error) {
+	False := false // I did it in order to get pointer to false
```

Comment:
I think there is no need to do that. As we discussed at the meeting, the default value and false can be the same.